test/test_child-pool.js

```diff
@@ -142,4 +142,20 @@ describe('Child pool', function() {
         expect(children).to.include(child);
       });
   });
+
+  it('should kill child after processing is finished and not retain it', function() {
+    var processor = __dirname + '/fixtures/fixture_processor_bar.js';
+    var child;
+
+    pool.setKeepProcesses(false);
+
+    return pool.retain(processor).then(function(_child) {
+      expect(_child).to.be.ok;
+      child = _child;
```

Comment:
Why this assignment?

Reply:
Must have been from copy-paste, you are right no need for re-assigned the variable, could `pool.release(_child);` directly

Reply:
Updated PR, removed the unnecessary reassignment